### PR TITLE
remove BlockDAO() from blockchain interface

### PR DESCRIPTION
**Pull request by koseoyoung**
following to PR #1783, remove BlockDAO() from blockchain interface and use dao directly (not through bc)

### CODE REVIEW

File: gasstation/gasstattion.go [hunk 1]
```diff
@@ -22,14 +23,16 @@ import (
 type GasStation struct {
 	bc  blockchain.Blockchain
 	sf  factory.Factory
+	dao blockdao.BlockDAO
```
Comment (CoderZhi) [inline gasstation/gasstattion.go]: let's define an interface here:
```
type BlockDAO interface {
  GetBlockHash,
  GetBlockByHeight,
}
```

File: blocksync/blocksync.go [hunk 2]
```diff
@@ -71,6 +72,7 @@ type blockSyncer struct {
 	buf              *blockBuffer
 	worker           *syncWorker
 	bc               blockchain.Blockchain
+	dao              blockdao.BlockDAO
```
Comment (CoderZhi) [inline blocksync/blocksync.go]: define interface:
```
type BlockDAO interface {
  GetBlockByHeight,
}
```

File: gasstation/gasstattion.go (hunk 1 above)
Comment (CoderZhi) [reply gasstation/gasstattion.go]: similarly, we can define an interface
```
type ExecutionSimulator interface {
  SimulateExecution,
}
```
or pass in the function SimulateExecution as a parameter of NewGasStation

Comment (koseoyoung) [reply gasstation/gasstattion.go]: fixed in latest commit

File: blocksync/blocksync.go (hunk 2 above)
Comment (koseoyoung) [reply blocksync/blocksync.go]: fixed in latest commit

**Pull request merged**